classes/plugins/ThemePlugin.inc.php

```diff
@@ -28,107 +53,364 @@ function ThemePlugin() {
+			return $style;
+		}
+
+		// If no parent theme, no style was found
+		if (!isset($this->parent)) {
+			return;
+		}
+
+		return $this->parent->getStyle($name);
+	}
+
+	/**
+	 * Add a script to load with this theme
+	 *
+	 * @param $name string A name for this script
+	 * @param $script string The script to be included. Should be path relative
+	 *   to the theme or, if the `inline` argument is included, script data to
+	 *   be output.
+	 * @param $args array Optional arguments hash. Supported args:
+	 *   `context` string Whether to load this on the `frontend` or `backend`.
+	 *      default: frontend
+	 *   `priority` int Controls order in which scripts are printed
+	 *      default: STYLE_SEQUENCE_NORMAL
+	 *   `inline` bool Whether the $script value should be output directly as
+	 *      script data. Used to pass backend data to the scripts.
+	 */
+	public function addScript($name, $script, $args = array()) {
+
+		if (!empty($args['inline'])) {
+			$args['script'] = $script;
+		} elseif (isset($args['baseUrl'])) {
+			$args['script'] = $args['baseUrl'] . $script;
+		} else {
+			$args['script'] = $this->_getBaseUrl($script);
+		}
+
+		$this->scripts[$name] = $args;
+	}
+
+	/**
+	 * Modify the params of an existing script
+	 *
+	 * @param $name string The name of the script to modify
+	 * @param $args array Parameters to modify.
+	 * @see self::addScript()
+	 * @return null
+	 */
+	public function modifyScript($name, $args = array()) {
+
+		$script = &$this->getScript($name);
+
+		if (empty($script)) {
+			return;
+		}
+
+		if (isset($args['path'])) {
+			$args['path'] = $this->_getBaseUrl($args['path']);
+		}
+
+		$script = array_merge( $script, $args );
 	}
 
 	/**
-	 * Get the filename to this theme's stylesheet, or null if none.
-	 * @return string|null
+	 * Get a script from this theme or any parent theme
+	 *
+	 * @param $name string The name of the script to retrieve
+	 * @return array|null Reference to the script or null if not found
 	 */
-	function getLessStylesheet() {
-		return null;
+	public function &getScript($name) {
+
+		// Search this theme
+		if (isset($this->scripts[$name])) {
+			$style = &$this->scripts[$name];
+			return $style;
+		}
+
+		// If no parent theme, no script was found
+		if (!isset($this->parent)) {
+			return;
+		}
+
+		return $this->parent->getScript($name);
 	}
 
 	/**
-	 * Get the compiled CSS cache filename
-	 * @return string|null
+	 * Set a parent theme for this theme
+	 *
+	 * @param $parent string Key in the plugin registry for the parent theme
+	 * @return null
 	 */
-	function getStyleCacheFilename() {
-		// Only relevant if Less compilation is used; otherwise return null.
-		if ($this->getLessStylesheet() === null) return null;
+	public function setParent($parent) {
 
-		return 'compiled-' . $this->getName() . '.css';
+		$parent = PluginRegistry::getPlugin('themes', $parent);
+
+		if (!is_a($parent, 'ThemePlugin')) {
+			return;
+		}
+
+		$this->parent = &$parent;
```

Comment:
No big deal, but the reference isn't necessary here (since PHP5 treats objects as references by default), and `=&` is the preferred syntax when references are used, rather than `= &`.